python/pyspark/sql/connect/session.py

```diff
@@ -395,13 +397,22 @@ def createDataFrame(
 
     createDataFrame.__doc__ = PySparkSession.createDataFrame.__doc__
 
-    def sql(self, sqlQuery: str, args: Optional[Dict[str, Any]] = None) -> "DataFrame":
-        cmd = SQL(sqlQuery, args)
-        data, properties = self.client.execute_command(cmd.command(self._client))
-        if "sql_command_result" in properties:
-            return DataFrame.withPlan(CachedRelation(properties["sql_command_result"]), self)
-        else:
-            return DataFrame.withPlan(SQL(sqlQuery, args), self)
+    def sql(
+        self, sqlQuery: str, args: Optional[Dict[str, Any]] = None, **kwargs: Any
+    ) -> "DataFrame":
+        formatter = SQLStringFormatter(self)
+        if len(kwargs) > 0:
+            sqlQuery = formatter.format(sqlQuery, **kwargs)
+        try:
+            cmd = SQL(sqlQuery, args, formatter.unresolvedColumns)
+            data, properties = self.client.execute_command(cmd.command(self._client))
+            if "sql_command_result" in properties:
+                return DataFrame.withPlan(CachedRelation(properties["sql_command_result"]), self)
+            else:
+                return DataFrame.withPlan(SQL(sqlQuery, args, formatter.unresolvedColumns), self)
```

Comment:
I meant sth like this
```python
            if "sql_command_result" in properties:
                ret = DataFrame.withPlan(CachedRelation(properties["sql_command_result"]), self)
            else:
                ret = DataFrame.withPlan(SQL(sqlQuery, args, formatter.unresolvedColumns), self)
            if len(kwargs) > 0:
                ret.schema  # Workaround to drop view blah blah blah by trggering analysis. blah blah
            return ret
```

Reply:
😂 It not work after tried, pyspark maybe worked because it will change plan after called.  But connect will not change plan in dataframe